src/JustEat.StatsD/V2/StatsDUtf8Formatter.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Text;
+
+namespace JustEat.StatsD.V2
+{
+    internal class StatsDUtf8Formatter
+    {
+        private readonly byte[] _utf8Prefix;
+
+        public StatsDUtf8Formatter(string prefix = "")
+        {
+            _utf8Prefix = string.IsNullOrWhiteSpace(prefix) ? new byte[0] : Encoding.UTF8.GetBytes(prefix + ".");
+        }
+
+        public int GetBufferSize(in StatsDMessage msg)
+        {
+            const int maxSerializedDoubleSymbols = 20;
+            const int maxUtf8BytesPerChar = 4;
+            const int colonBytes = 1;
+
+            const int maxMessageKindSuffixSize = 3;
+            const int maxSamplingSuffixSize = 2;
+
+            return _utf8Prefix.Length + msg.StatBucket.Length * maxUtf8BytesPerChar
+              + colonBytes
+              + maxSerializedDoubleSymbols
+              + maxMessageKindSuffixSize
+              + maxSamplingSuffixSize
+              + maxSerializedDoubleSymbols;
+        }
+
+        // prefix + msg.Bucket + ":" + msg.Value + (<oneOf> "|ms", "|c", "|g") + {<optional> "|@" + sampleRate }
+        public bool TryFormat(in StatsDMessage msg, double sampleRate, Span<byte> destination, out int written)
+        {
+            written = 0;
+            var buffer = new Buffer(destination);
+
+            if (!buffer.TryWriteBytes(_utf8Prefix)) return false;
```

Comment:
Braces please.

Reply:
would add 50 lines of code, would definitely make it harder to read. I dig the default coding guidelines. But here `if (false) return` is more like F# `do!` (monadic bind) in a sense that it terminates computation,

Reply:
I'll update if you insist though